tests/unit/broccoli/ember-app-test.js

```diff
@@ -938,20 +939,188 @@ describe('broccoli/ember-app', function() {
+        // "wiring" not the implementation of our dependencies e.g. broccoli-concat
+        app._concatFiles = function(tree, options) {
+          count++
+          args.push(options);
+          return tree;
+        };
+
+        app.appAndDependencies = function() {
+          return 'app-and-dependencies-tree';
+        };
+      });
+
+      it('correctly orders concats from app.styles()', function() {
+        app.import('files/c.css');
+        app.import('files/b.css');
+        app.import('files/a.css', { prepend: true });
+        app.import('files/a.css');
+
+        app.styles(); // run
+
+        expect(count).to.eql(3);
+
+        expect(args[0]).to.deep.eql({
+          allowNone: true,
+          annotation: 'Concat: Addon CSS',
+          inputFiles: [
+            '**/*.css'
+          ],
+          outputFile: '/addons.css'
+        });
+
+        expect(args[1]).to.deep.eql({
+          allowNone: true,
+          annotation: 'Concat: Addon JS',
+          inputFiles: [
+            '**/*.js'
+          ],
+          outputFile: '/addons.js'
+        });
+
+        expect(args[2]).to.deep.eql({
+          annotation: 'Concat: Vendor Styles/assets/vendor.css',
+          headerFiles: [
+            'files/a.css',
+            'files/c.css',
+            'files/b.css',
+            'files/a.css',
+            'vendor/addons.css',
+          ],
+          outputFile: '/assets/vendor.css'
+        });
+      });
+      it('correct orders concats from app.javacsript()', function() {
+        app.import('files/c.js');
+        app.import('files/b.js');
+        app.import('files/a.js', { prepend: true });
+        app.import('files/a.js');
+
+        app.javascript(); // run
+
+        expect(count).to.eql(2);
+        // should be unrelated files
+        expect(args[0]).to.deep.eql({
+          annotation: "Concat: App",
+          footerFiles: [
+            "vendor/ember-cli/app-suffix.js",
+            "vendor/ember-cli/app-config.js",
+            "vendor/ember-cli/app-boot.js"
+          ],
+          headerFiles: [
+            "vendor/ember-cli/app-prefix.js"
+          ],
+          inputFiles: [
+            "test-project/**/*.js"
+          ],
+          outputFile: "/assets/test-project.js"
+        });
+
+        // should be: a,c,b,a in output
+        expect(args[1]).to.deep.eql({
+          annotation: "Concat: Vendor /assets/vendor.js",
+          headerFiles: [
+            "vendor/ember-cli/vendor-prefix.js",
+            "files/a.js",
+            "bower_components/jquery/dist/jquery.js",
+            "bower_components/ember/ember.js",
+            "bower_components/ember-cli-shims/app-shims.js",
+            "files/c.js",
+            "files/b.js",
+            "files/a.js",
+            "vendor/addons.js",
+            "vendor/ember-cli/vendor-suffix.js"
+          ],
+          outputFile: "/assets/vendor.js",
+          separator: EOL + ';',
```

Comment:
@Turbo87 we actually explicitly specify EOL [here](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L1272). But we may want to investigate this as it doesn't seem right. As this merely tests existing behavior, should we consider the change as part of this PR or a subsequent. I can be convinced of either